### PR TITLE
Add ref_particle charge & mass as user options in dashboard

### PR DESCRIPTION
**Pull request by proy30**
Dashboard currently does not support this and has ref. particle charge/mass hard-coded to -1.0 and 0.510998950 respectively.

- [X] Add to UI
- [X] Connect to Simulation
- [x] Validation:
  - For charge: Int & non-zeor
  - For mass: Greater than 0
- [X] Update export_template

### CODE REVIEW

**Comment by proy30** [comment]
UI look - 9/29[
chrome_O6TlmYtDoq
]

File: src/python/impactx/dashboard/Analyze/plot_PhaseSpaceProjections/phaseSpace.py [hunk 1]
```diff
@@ -53,7 +53,9 @@ def run_simulation():
     #   reference particle
```
Comment (ax3l) [inline src/python/impactx/dashboard/Analyze/plot_PhaseSpaceProjections/phaseSpace.py:53]: In a follow-up PR, `run_simulation()` really needs to be moved into a more well-named file than `Analyze/plot_PhaseSpaceProjections/phaseSpace.py` please :)

**Review by ax3l — APPROVED** [review]
Nice, looks good!

File: src/python/impactx/dashboard/Analyze/plot_PhaseSpaceProjections/phaseSpace.py (hunk 1 above)
Comment (proy30) [reply src/python/impactx/dashboard/Analyze/plot_PhaseSpaceProjections/phaseSpace.py:53]: Agree, addressed in https://github.com/ECP-WarpX/impactx/pull/725!

**Pull request merged**